### DISCUSSION TITLE
Various questions about transaction events

### DISCUSSION THREAD

**Discussion by deepisidhu** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#221] [General]
Hello, I have a few questions about different transactions:

1. What is happening during a security release transaction? Is this only on RSU's?
`      "object_type": "TX_PLAN_SECURITY_RELEASE",
      "id": "test-plan-security-release-full-fields",
      "security_id": "387878ba-8fb6-4673-812e-32c092947899",
      "date": "2017-07-22",
      "settlement_date": "2017-07-22",
      "release_price": {
        "amount": "9.00",
        "currency": "CAD"
      },
      "comments": ["Release the securities"],
      "net_quantity": "42",
      "method": "Method of release",
      "stock_swap": false,
      "cash_paid": {
        "amount": "0.00",
        "currency": "CAD"
      },
      "quantity_sold": "1",
      "sale_price_per_unit": {
        "amount": "0.00",
        "currency": "CAD"
      },
      "withheld_quantity": "0"
    },`

2. What's the difference between the purchase price and exercise price in a Warrant Issuance?
`  {
      "object_type": "TX_WARRANT_ISSUANCE",
      "id": "test-warrant-issuance-minimal",
      "security_id": "test-security-id",
      "date": "2022-02-01",
      "custom_id": "S-1",
      "stakeholder_id": "stakeholder-id",
      "security_law_exemptions": [],
      "quantity": "1000",
      "exercise_price": {
        "amount": "1.00",
        "currency": "USD"
      },
      "purchase_price": {
        "amount": "1.00",
        "currency": "USD"
      },
      "exercise_triggers": [
        {
          "trigger_id": "WARRANT-1.TRIG.1",
          "nickname": "Exercisable Until Expiration",
          "trigger_description": "The warrant is exercisable on or before its date of expiration",
          "trigger_type": "ELECTIVE_ON_OR_BEFORE_DATE",
          "trigger_date": "2029-01-01",
          "conversion_right": {
            "conversion_mechanism": {
              "mechanism_type": "FIXED_AMOUNT_CONVERSION",
              "converts_to_quantity": "10000.00"
            },
            "converts_to_stock_class_id": "stock-class-id"
          }
        }
      ],
      "warrant_expiration_date": "2032-02-01"
    },`

3. For Convertible security types, is there a difference in the types of attributes expected for a NOTE, SAFE, and CONVERTIBLE_SECURITY?

**Comment by pjohnmeyer** [comment]
@deepisidhu 

> What is happening during a security release transaction? Is this only on RSU's?

The only note I have on this is from a draft document, and it reads as fairly circular to me:

"When the right of repurchase is actually removed from the PlanSecurity. For use in situations where some action is required to free the plan security from repurchase right" @JSv4 can you shine a light on this?

> What's the difference between the purchase price and exercise price in a Warrant Issuance?

From the documentation:
`purchase_price`: "Actual purchase price of the warrant (sum up purported value of all consideration, including in-kind)"
`exercise_price`: "The exercise price of the warrant"

This latter one is not terribly helpful, but should probably be modified to say that it is the price _per share_, whereas the `purchase_price` is the total purchase price, which may include other considerations than the per share price.

> For Convertible security types, is there a difference in the types of attributes expected for a NOTE, SAFE, and CONVERTIBLE_SECURITY?

I don't see anything in the schema that changes the expected data based on this value. Do you think there needs to be?
